### PR TITLE
Update Tracing to use Opentelemetry

### PR DESCRIPTION
**Pull request by hananiel**
With this PR ExportersBase and ExportersCore got removed because 1. CloudfoundryForwarder is now deprecated. 2. Zipkin exporter is now wired in the builder and all of the Steeltoe implementation is obsoleted in favor of using Opentelemetry implementation. The only thing left is the options class and interface which i moved to the OpentelemetryBase.  Instead of the

```
services.AddZipkinExporter(Configuration) 
... 
app.UseTracingExporter() 
```
We will now add zipkin or other exporters like this: 
```
services.AddDistributedTracing(Configuration, builder => builder.UseZipkinWithTraceOptions(services));
```

### CODE REVIEW

File: src/Management/src/TracingBase/Steeltoe.Management.TracingBase.csproj [hunk 1]
```diff
@@ -2,25 +2,24 @@
...
   </ItemGroup>
 
   <ItemGroup Condition="'$(CI_BUILD)' == 'True'">
     <PackageReference Include="Steeltoe.Extensions.Logging.Abstractions" Version="$(SteeltoeVersion)$(SteeltoeVersionSuffix)" />
     <PackageReference Include="Steeltoe.Management.Abstractions" Version="$(SteeltoeVersion)$(SteeltoeVersionSuffix)" />
-    <PackageReference Include="Steeltoe.Management.OpenCensusBase" Version="$(SteeltoeVersion)$(SteeltoeVersionSuffix)" />
+    <PackageReference Include="Steeltoe.Management.OpenTelemetryBase" Version="$(SteeltoeVersion)$(SteeltoeVersionSuffix)" />
+  </ItemGroup>
+
+  <ItemGroup>
```
Comment (TimHess) [inline src/Management/src/TracingBase/Steeltoe.Management.TracingBase.csproj]: This should be either package or project reference like the others above

File: src/Management/src/EndpointBase/Steeltoe.Management.EndpointBase.csproj [hunk 2]
```diff
@@ -42,4 +41,8 @@
     <PackageReference Include="Steeltoe.Management.Diagnostics" Version="$(SteeltoeVersion)$(SteeltoeVersionSuffix)" />
     <PackageReference Include="Steeltoe.Management.OpenCensusBase" Version="$(SteeltoeVersion)$(SteeltoeVersionSuffix)" />
   </ItemGroup>
+  
+  <ItemGroup>
```
Comment (TimHess) [inline src/Management/src/EndpointBase/Steeltoe.Management.EndpointBase.csproj]: This should be either package or project reference like the other Steeltoe refs

File: nuget.config [hunk 3]
```diff
@@ -4,5 +4,6 @@
     <add key="SteeltoeDev" value="https://www.myget.org/F/steeltoedev/api/v3/index.json" />
     <add key="NuGet" value="https://api.nuget.org/v3/index.json" />
     <add key="DotnetCore" value="https://dotnet.myget.org/F/dotnet-core/api/v3/index.json" />
+    <add key="OpenTelemetryDev" value="https://www.myget.org/F/opentelemetry/api/v3/index.json" />
```
Comment (TimHess) [inline nuget.config]: Is there substantial difference between the package available on nuget.org and what's available here? If we ship with a dependency from MyGet, others will need to add this to their nuget.config until we change to a version that's on nuget.org. If we need to have this, it should be clearly documented and have a follow-up issue logged somewhere

File: src/Management/Management.sln [hunk 4]
```diff
@@ -29,32 +29,16 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Steeltoe.Management.CloudFo
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Steeltoe.Management.CloudFoundryCore.Test", "test\CloudFoundryCore.Test\Steeltoe.Management.CloudFoundryCore.Test.csproj", "{B94F21CC-3A87-4D9A-8F9F-39BC4B7447D1}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Steeltoe.Management.EndpointCore.Test", "test\EndpointCore.Test\Steeltoe.Management.EndpointCore.Test.csproj", "{D1E2C1E0-8DF9-403B-A8C8-81C2A259B563}"
```
Comment (TimHess) [inline src/Management/Management.sln]: Was this removal intentional?

File: src/Management/src/OpenTelemetryBase/Steeltoe.Management.OpenTelemetryBase.csproj [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    <RootNamespace>Steeltoe.Management.OpenTelemetry</RootNamespace>
+  </PropertyGroup>
+  <Import Project="..\..\..\..\sharedproject.props" />
+
+  <ItemGroup>
+    <PackageReference Include="OpenTelemetry" Version="$(OpenTelemetryVersion)" />
+    <PackageReference Include="OpenTelemetry.Exporter.Zipkin" Version="$(OpenTelemetryVersion)" />
+  </ItemGroup>
+
+  <ItemGroup>
```
Comment (TimHess) [inline src/Management/src/OpenTelemetryBase/Steeltoe.Management.OpenTelemetryBase.csproj]: use `Condition="'$(CI_BUILD)' == 'True'"` to make this conditionally a package or project reference

File: src/Management/src/TracingCore/Observer/AspNetCoreHostingObserver.cs [hunk 6]
```diff
@@ -209,23 +193,14 @@ protected internal string ExtractSpanName(HttpContext context)
...
-                    return result;
-                });
-            }
-            catch (SpanContextParseException)
-            {
-                // Ignore
-            }
-
-            return null;
+            return Propagation.Extract(request.Headers, (d, k) =>
```
Comment (TimHess) [inline src/Management/src/TracingCore/Observer/AspNetCoreHostingObserver.cs]: Will this change of removing the try/catch result in unhandled exceptions if there are failures in here?

File: src/Management/src/OpenTelemetryBase/Trace/SpanExtensions.cs [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+
+using OpenTelemetry.Trace;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+
+namespace Steeltoe.Management.OpenTelemetry.Trace
+{
+    public static class SpanExtensions
+    {
+        /*
```
Comment (TimHess) [inline src/Management/src/OpenTelemetryBase/Trace/SpanExtensions.cs]: Might want to add a comment about why all this code is commented out

File: src/Management/src/TracingCore/Steeltoe.Management.TracingCore.csproj [hunk 8]
```diff
@@ -15,10 +15,12 @@
 
   <ItemGroup Condition="'$(CI_BUILD)' == ''">
     <ProjectReference Include="..\..\..\Logging\src\DynamicLogger\Steeltoe.Extensions.Logging.DynamicLogger.csproj" />
-    <ProjectReference Include="..\TracingBase\Steeltoe.Management.TracingBase.csproj" />
   </ItemGroup>
   <ItemGroup Condition="'$(CI_BUILD)' == 'True'">
     <PackageReference Include="Steeltoe.Extensions.Logging.DynamicLogger" Version="$(SteeltoeVersion)$(SteeltoeVersionSuffix)" />
     <PackageReference Include="Steeltoe.Management.TracingBase" Version="$(SteeltoeVersion)$(SteeltoeVersionSuffix)" />
   </ItemGroup>
+  <ItemGroup>
```
Comment (TimHess) [inline src/Management/src/TracingCore/Steeltoe.Management.TracingCore.csproj]: Changes to this file should be reverted

File: src/Management/test/TracingBase.Test/Steeltoe.Management.TracingBase.Test.csproj [hunk 9]
```diff
@@ -8,6 +8,7 @@
   <ItemGroup>
     <ProjectReference Include="..\..\..\Common\src\Common\Steeltoe.Common.csproj" />
     <ProjectReference Include="..\..\src\TracingBase\Steeltoe.Management.TracingBase.csproj" />
+
```
Comment (TimHess) [inline src/Management/test/TracingBase.Test/Steeltoe.Management.TracingBase.Test.csproj]:
```suggestion
```

File: src/Management/src/TracingCore/Observer/AspNetCoreHostingObserver.cs (hunk 6 above)
Comment (hananiel) [reply src/Management/src/TracingCore/Observer/AspNetCoreHostingObserver.cs]: No because the library caches and returns an invalid Span. Having that was giving an unreachable code warning.

File: nuget.config (hunk 3 above)
Comment (hananiel) [reply nuget.config]: I changed it to nuget version

File: src/Management/src/TracingBase/Steeltoe.Management.TracingBase.csproj (hunk 1 above)
Comment (hananiel) [reply src/Management/src/TracingBase/Steeltoe.Management.TracingBase.csproj]: done

File: src/Management/src/OpenTelemetryBase/Steeltoe.Management.OpenTelemetryBase.csproj (hunk 5 above)
Comment (hananiel) [reply src/Management/src/OpenTelemetryBase/Steeltoe.Management.OpenTelemetryBase.csproj]: done

File: src/Management/src/OpenTelemetryBase/Trace/SpanExtensions.cs (hunk 7 above)
Comment (hananiel) [reply src/Management/src/OpenTelemetryBase/Trace/SpanExtensions.cs]: done

File: src/Management/src/TracingCore/Steeltoe.Management.TracingCore.csproj (hunk 8 above)
Comment (hananiel) [reply src/Management/src/TracingCore/Steeltoe.Management.TracingCore.csproj]: done

File: src/Management/test/TracingBase.Test/Steeltoe.Management.TracingBase.Test.csproj (hunk 9 above)
Comment (hananiel) [reply src/Management/test/TracingBase.Test/Steeltoe.Management.TracingBase.Test.csproj]: done
